### PR TITLE
Stack

### PR DESCRIPTION
**Pull request by constanthatz**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 This repository holds sample code for a number of data structures implemented with python.
 
 20150204 - Linked list add
+20150205 - Stack add
```
Comment (cewing) [inline README.md]: I want better descriptions of what this all does.  Write me up some documentation.

File: stack.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+
+class Stack(object):
+    ''' Create an empty stack. '''
+    def __init__(self):
+        self.top = None
+
+    def push(self, element):
+        ''' Add data element to the top of stack. '''
+        self.top, self.top.previous = element, self.top
```
Comment (cewing) [inline stack.py]: As I stated in class, we changed this to require that you put a value in and get a value out.  Could you please update it?

File: stack.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env python
+
+
+class Element(object):
+    ''' Create data element with default value and previous pointer. '''
+    def __init__(self, value=None, previous=None):
+        ''' Value and previous pointer default to none. '''
+        self.val = value
+        self.previous = previous
```
Comment (cewing) [inline stack.py:10]: Is there ever any reason to create an element with no value?  If not, why make it an optional argument.

File: stack.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        self.top = None
+
+    def push(self, element):
+        ''' Add data element to the top of stack. '''
+        self.top, self.top.previous = element, self.top
+
+    def pop(self):
+        ''' Remove top element from stack. Reassign and reassign top data
+            element. '''
+        if not self.top:
```
Comment (cewing) [inline stack.py]: This is a place where "it is better to ask forgiveness than permission" could come in handy.  Don't bother checking if self.top exists, just try to get the value and deal with the `AttributeError` if it doesn't.

File: test_stack.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
+from __future__ import unicode_literals
+import pytest
```
Comment (cewing) [inline test_stack.py:2]: if you don't use any specifics of the `pytest` module, you need not import it.  here you aren't using it at all.

Comment (constanthatz) [inline test_stack.py:2]: We use pytest.raises.

Comment (cewing) [inline test_stack.py:2]: @constanthatz: shoot.  missed it.  Sorry

File: README.md [hunk 6]
```diff
@@ -1,15 +1,35 @@
...
-tutorialspoint.com/python
 
+Resources:
+<ul>
+<li>Cris Ewing</li>
+<li>Mark Ableidinger</li>
+<li>stackoverflow.com</li>
+<li>tutorialspoint.com/python</li>
+<li>docs.python.org</li>
+</ul>
```
Comment (cewing) [inline README.md:35]: gah,  don't write html in markdown.  just write proper markdown.  Work at it until you get it right.

**Pull request merged**